gitops/argocd/tekton-results/deployment.yaml

```diff
@@ -0,0 +1,111 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  labels:
+    app.kubernetes.io/component: api
+    app.kubernetes.io/name: tekton-results
+    app.kubernetes.io/part-of: tekton-results
+    app.kubernetes.io/version: v0.3.0
+    results.tekton.dev/release: devel
+    version: devel
+  name: tekton-results-api
+  namespace: openshift-pipelines
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: tekton-results-api
+      app.kubernetes.io/part-of: tekton-results
+      app.kubernetes.io/version: v0.3.0
+  template:
+    metadata:
+      annotations:
+        cluster-autoscaler.kubernetes.io/safe-to-evict: "false"
+      labels:
+        app: tekton-results-api
+        app.kubernetes.io/component: api
+        app.kubernetes.io/name: tekton-results
+        app.kubernetes.io/part-of: tekton-results
+        app.kubernetes.io/version: v0.3.0
+        results.tekton.dev/release: devel
+        version: devel
+    spec:
+      containers:
+        - env:
+            - name: DB_USER
+              valueFrom:
+                secretKeyRef:
+                  key: POSTGRES_USER
+                  name: tekton-results-postgres
+            - name: DB_PASSWORD
+              valueFrom:
+                secretKeyRef:
+                  key: POSTGRES_PASSWORD
+                  name: tekton-results-postgres
+            - name: DB_PROTOCOL
+              value: tcp
+            - name: DB_ADDR
+              value: postgres-service.openshift-pipelines.svc.cluster.local
+            - name: DB_NAME
+              value: tekton-results
+          image: gcr.io/tekton-releases/github.com/tektoncd/results/cmd/api:v0.3.0@sha256:ed956232dc782cb9bb16f7bf15459fd56db52ce0952b5e2be70ae5314d0ad8de
+          name: tekton-results-api
+          volumeMounts:
+            - mountPath: /etc/tls
+              name: tls
+              readOnly: true
+      serviceAccountName: tekton-results-api
+      volumes:
+        - name: tls
+          secret:
+            secretName: tekton-results-tls
```

Comment:
Noticing that our GitOps approach doesn't make syncing the TLS secret easy. However, we do have access to Cert-manager as an operator, which makes it simple to create declarative, self-signed certs. We'd configure this as follows:

- Add a Subscription to openshift-cert-manager-operator - probably under `gitops/argocd/cert-manager` (with a corresponding Application in `argo-apps`)
- Create a Self-signed Issuer  https://cert-manager.io/docs/configuration/selfsigned/
- Create a Certificate using the self-signed issuer: https://cert-manager.io/docs/usage/certificate/

Reply:
@adambkaplan We are already using cert-manager in ckcp [here](https://github.com/openshift-pipelines/pipelines-service/tree/main/ckcp/openshift-operators/cert-manager). If we need to extend this for results as well, we could create a folder called operators under the root directory and place cert-manager, gitops operators there. In the future, we can extend that to other operators as well.

Reply:
@bnallapeta if I am understanding correctly cert-manager becomes part of the platform if it is used for Tekton results. As such it should be deployed through gitops.
@adambkaplan self-signed certificates are not really suitable for production use if the endpoint is accessed by a user or an external system. Is the idea that the endpoint would be accessed through the gateway / kcp loadbalancer configured with reencrypt, which would then have a certificate signed by a trusted authority and trust the CA used by cert-manager?

Reply:
@fgiloux The certs are used by both the grpc and rest service that will be part of tekton-results. This is expected to be accessible by the user behind proper routes with reencrypt enabled. 

is cert-manager part of the deployment on all clusters or only as part of ckcp installs ?

Reply:
> @adambkaplan self-signed certificates are not really suitable for production use if the endpoint is accessed by a user or an external system. Is the idea that the endpoint would be accessed through the gateway / kcp loadbalancer configured with reencrypt, which would then have a certificate signed by a trusted authority and trust the CA used by cert-manager?

The gRPC server will likely need to be accessed through the gateway or equivalent REST service, in which case yes we would have a cert signed with a trusted CA. @sayan-biswas is researching this.

Reply:
@CathalOConnorRH Currently, it is part of just ckcp. But if we need to make it part of Tekton Results, then we could do [this](https://github.com/openshift-pipelines/pipelines-service/pull/139#discussion_r919682468) and deploy it via GitOps as @fgiloux has mentioned.

Reply:
@bnallapeta I've opened a PR to move cert-manager install [here](https://github.com/openshift-pipelines/pipelines-service/pull/163)

Marking this as draft as there are external requirements and further changes needed to this PR before tekton-results can be merged.